tests/suite/graceful_recovery_test.go

```diff
@@ -100,8 +102,94 @@ var _ = Describe("Graceful Recovery test", Ordered, Label("functional", "gracefu
 		Skip("Test currently fails due to this issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/1108")
 		runRecoveryTest(teaURL, coffeeURL, ngfPodName, nginxContainerName, files, &ns)
 	})
+
+	It("recovers when drained node is restarted", func() {
+		runRestartNodeWithDrainingTest(teaURL, coffeeURL, files, &ns)
+	})
+
+	It("recovers when node is restarted abruptly", func() {
+		// FIXME(bjee19) remove Skip() when https://github.com/nginxinc/nginx-gateway-fabric/issues/1108 is completed.
+		Skip("Test currently fails due to this issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/1108")
+		runRestartNodeAbruptlyTest(teaURL, coffeeURL, files, &ns)
+	})
 })
 
+func runRestartNodeWithDrainingTest(teaURL, coffeeURL string, files []string, ns *core.Namespace) {
+	runRestartNodeTest(teaURL, coffeeURL, files, ns, true)
+}
+
+func runRestartNodeAbruptlyTest(teaURL, coffeeURL string, files []string, ns *core.Namespace) {
+	runRestartNodeTest(teaURL, coffeeURL, files, ns, false)
+}
+
+func runRestartNodeTest(teaURL, coffeeURL string, files []string, ns *core.Namespace, drain bool) {
+	nodeNames, err := getNodeNames()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(nodeNames).To(HaveLen(1))
+
+	kindNodeName := nodeNames[0]
+
+	if portFwdPort != 0 {
+		close(portForwardStopCh)
+	}
+
+	if drain {
+		_, err := exec.Command(
+			"kubectl",
+			"drain",
+			kindNodeName,
+			"--ignore-daemonsets",
+			"--delete-local-data",
+		).CombinedOutput()
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = exec.Command("kubectl", "delete", "node", kindNodeName).CombinedOutput()
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	containerName := *clusterName + "-control-plane"
```

Comment:
Should we add a nil check for clusterName?

Reply:
We still need a nil check before we dereference clusterName on line 144:

```
Expect(clusterName).ToNot(BeNil(), "clusterName variable not set")
```

Reply:
Ohh I see that makes more sense, is it overkill to have the nil check and the empty string check?

Reply:
Good question. I wonder if instead of checking for nil, just check that it's not empty without dereferencing:

```
Expect(clusterName).ToNot(BeEmpty(), "clusterName variable not set")
```

does that work?

Reply:
no it doesn't seem to work. 

```
[FAILED] clusterName variable not set
  BeEmpty matcher expects a string/array/map/channel/slice.  Got:
      <*string | 0x140005a68d0>: 
```

Reply:
bummer. Then I think we need to add both checks